optimize/dataio.py

```diff
@@ -13,7 +13,7 @@ def structured_from_torch(tracks_torch, dtype):
     return rfn.unstructured_to_structured(tracks_torch.cpu().numpy(), dtype=dtype)
 
 class TracksDataset(Dataset):
-    def __init__(self, filename, ntrack, swap_xz=True, seed=3, random_ntrack=False, track_zlen_sel=2., 
+    def __init__(self, filename, ntrack, max_nbatch, iterations, swap_xz=True, seed=3, random_ntrack=False, track_zlen_sel=2., 
```

Comment:
Small style thing, but I'd prefer if the dataset didn't need any info about the fit iterations -- you can handle the iterations/max_nbatch logic before passing max_nbatch here